qiskit/quantum_info/operators/unitary.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=invalid-name,assignment-from-no-return
+
+"""
+Tools for working with Unitary Operators.
+
+A simple unitary class and some tools.
+"""
+import copy
+import numpy
+import sympy
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.gate import Gate
+from qiskit.dagcircuit import DAGCircuit
+
+
+class Unitary(Gate):
+    """Class for representing unitary operators"""
+
+    def __init__(self, matrix, *qargs, label=None, validate=True, rtol=1e-5, atol=1e-8):
+        """
+        Create unitary.
+
+        Args:
+            matrix (numpy.ndarray or list(list)): unitary matrix representation.
+            qargs (list(tuple(QuantumCircuit, int))): qubits to apply operation to.
+            label (str): identifier hint for backend
```

Comment:
I'm not sure what the use case for a `label` is. This class derives from `Gate`, so it already has a `name`. Isn't that just the label?

Reply:
ah i see. the `name` is always "unitary".